http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -143,11 +182,80 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         raw"persistence.storageResourceGroup=${cloudContext.managedResourceGroupName.value}",
         raw"persistence.storageAccount=${landingZoneResources.storageAccountName.value}",
 
+        // identity configs
+        raw"identity.name=${petManagedIdentity.name()}",
+        raw"identity.resourceId=${petManagedIdentity.id()}",
+        raw"identity.clientId=${petManagedIdentity.clientId()}",
+
         // general configs
         raw"fullnameOverride=coa-${release.asString}"
       ).mkString(",")
     )
 
+  private[util] def assignVmScaleSet(clusterName: AKSClusterName,
+                                     cloudContext: AzureCloudContext,
+                                     petManagedIdentity: Identity
+  )(implicit ev: Ask[F, AppContext]): F[Unit] = for {
+    // Resolve the cluster in Azure
+    cluster <- azureContainerService.getCluster(clusterName, cloudContext)
+
+    // Resolve the VM scale set backing the node pool
+    // Note: we are making the assumption here that there is 1 node pool per cluster.
+    compute <- buildComputeManager(cloudContext)
+    getFirstVmScaleSet = for {
+      vmScaleSets <- F.delay(compute.virtualMachineScaleSets().listByResourceGroup(cluster.nodeResourceGroup()))
+      vmScaleSet <- F
+        .fromOption(
+          vmScaleSets.iterator().asScala.nextOption(),
+          AppCreationException(
+            s"VM scale set not found for cluster ${cloudContext.managedResourceGroupName.value}/${clusterName.value}"
+          )
+        )
+    } yield vmScaleSet
+
+    // Retry getting the VM scale set since Azure returns an empty list sporadically for some reason
+    retryConfig = RetryPredicates.retryAllConfig
+    vmScaleSet <- tracedRetryF(retryConfig)(
+      getFirstVmScaleSet,
+      s"Get VM scale set for cluster ${cloudContext.managedResourceGroupName.value}/${clusterName.value}"
+    ).compile.lastOrError
+
+    // Assign VM scale set to the pet UAMI (if not already assigned).
+    //
+    // Note: normally this is done behind the scenes by aad-pod-identity. However in our case the deny assignments
+    // block it, so we need to use "Managed" mode handle the assignment ourselves. For more info see:
+    // https://azure.github.io/aad-pod-identity/docs/configure/standard_to_managed_mode/
```

Comment:
I knew about Workload Identity, I actually tried to use that first. See doc: https://docs.google.com/document/d/1ND30g6i62HZwPSG9RB1lPbZSukEWuqJyg3ioHo9xGQc/edit

I didn't know they'd deprecate aad-pod-identity 3 days ago. 😅 

I think Workload Identity is still probably the right end state for us, but there some hurdles trying to use it today -- namely a few key pieces of functionality were missing from the Java SDK (since it's technically a Preview feature). I met with some MSFT folks and they suggested using aad-pod-identity since it's all at the helm level, and it is also what the MSFT-maintained [cromwell on azure](https://github.com/microsoft/CromwellOnAzure/tree/develop/src/deploy-cromwell-on-azure/scripts/helm) is using.

Even so I'll make a ticket to migrate to Workload Identity in the future.

Reply:
https://broadworkbench.atlassian.net/browse/IA-3804